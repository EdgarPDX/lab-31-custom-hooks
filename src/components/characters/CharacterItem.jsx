import React from 'react';
import PropTypes from 'prop-types';

const CharacterItem = ({ name, occupation, image, description }) => (
  <figure>
    <figcaption>Name: {name}</figcaption>
    <img src={image} alt={name}/>
    <figcaption>Occupation: {occupation}</figcaption>
    <p>{description}</p>
  </figure>
);


CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default CharacterItem;
