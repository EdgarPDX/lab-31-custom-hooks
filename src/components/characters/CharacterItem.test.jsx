import React from 'react';
import { render } from '@testing-library/react';
import CharacterItem from './CharacterItem';


describe('Character component', () => {
  it('renders Character', () => {
    const { asFragment } = render(<CharacterItem
      name="Adam Pierce"
      occupation="Cult leader"
      image="https://vignette.wikia.nocookie.net/x-files/images"
      description="Scientist working at the California Institute of Technology."
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
