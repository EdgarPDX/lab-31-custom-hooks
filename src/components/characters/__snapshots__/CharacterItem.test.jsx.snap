// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <figcaption>
      Name: Adam Pierce
    </figcaption>
    <img
      alt="Adam Pierce"
      src="https://vignette.wikia.nocookie.net/x-files/images"
    />
    <figcaption>
      Occupation: Cult leader
    </figcaption>
    <p>
      Scientist working at the California Institute of Technology.
    </p>
  </figure>
</DocumentFragment>
`;
